File: src/main/java/com/github/thorben/webnd/domain/user/Password.java
package com.github.thorben.webnd.domain.user;

import com.github.thorben.webnd.foundation.AbstractValueObject;

import javax.persistence.Embeddable;

@Embeddable
public class Password extends AbstractValueObject<String> {

	protected Password() {
		// JPA
	}

	public Password(String value) {
		super(value);
	}
}

File: src/main/kotlin/com/github/thorben/webnd/domain/user/Password.kt
package com.github.thorben.webnd.domain.user

import com.github.thorben.webnd.foundation.AbstractValueObject
import javax.persistence.Embeddable

@Embeddable
class Password(value: String) : AbstractValueObject<String>(value) {
	protected constructor(): this("") {
		// JPA
	}
}